daisi/src/cpps/logical/algorithms/disposition/round_robin_initiator.cpp

```diff
@@ -0,0 +1,231 @@
+
+  // check wether all tasks have to be distributed, or only parts of it
+  if (previously_allocated) {
+    tasks = unallocated_tasks_per_mf_[mf_id];
+  } else {
+    //   tasks = material_flow.getTasks();
+    unallocated_tasks_per_mf_[mf_id] = tasks;
+  }
+  for (auto &task : tasks) {
+    assignTask(task);
+  }
+  ns3::Simulator::Schedule(ns3::Seconds(delays_.wait_to_receive_assignment_response),
+                           &RoundRobinInitiator::processAssignmentAcceptions, this, index);
+}
+
+void RoundRobinInitiator::assignTask(const material_flow::Task &task) {
+  auto chosen_participant = chooseParticipantForTask(task);
+
+  // cast ParticipantInfo into ParticipantInfoRoundRobin
+  std::shared_ptr<RoundRobinInitiator::ParticipantInfoRoundRobin> chosen_participant_rr =
+      std::static_pointer_cast<RoundRobinInitiator::ParticipantInfoRoundRobin>(chosen_participant);
+
+  // inform about task assignment
+  auto assignment_notification = AssignmentNotification(task, sola_->getConectionString());
+  sola_->sendData(serialize(assignment_notification),
+                  sola::Endpoint(chosen_participant_rr->connection_string));
+
+  // update participant and priority queue
+  chosen_participant_rr->assignment_counter++;
+  participants_per_ability_[chosen_participant_rr->ability].pop();
+  participants_per_ability_[chosen_participant_rr->ability].push(*chosen_participant_rr);
+}
+
+std::shared_ptr<CentralizedInitiator::ParticipantInfo>
+RoundRobinInitiator::chooseParticipantForTask(const material_flow::Task &task) {
+  // discover participants which can execute the task
+  auto fitting_abilities =
+      AmrFleet::get().getFittingExistingAbilities(task.getAbilityRequirement());
+  std::vector<ParticipantInfoRoundRobin> possible_candidates;
+  if (fitting_abilities.empty()) {
+    throw std::runtime_error("The task requirements cannot be met by any participant.");
+  }
+
+  for (const auto &ability : fitting_abilities) {
+    // observe the participant with the highest priority per ability
+    auto first_participant = participants_per_ability_[ability].top();
+
+    // check wether the priority is high enough to become a possible candidate for the task
+    if (possible_candidates.empty() ||
+        first_participant.assignment_counter <= possible_candidates[0].assignment_counter) {
+      if (!possible_candidates.empty() &&
+          first_participant.assignment_counter < possible_candidates[0].assignment_counter) {
+        // higher prio than all other participants so far
+        possible_candidates.clear();
+      }
+      possible_candidates.push_back(first_participant);
+    }
+  }
+  if (possible_candidates.empty()) {
+    throw std::runtime_error("No AMR matches the task requirements");
+  }
+  int random_index = 0;
+  if (possible_candidates.size() > 1) {
+    // choose AMR randomly
+    random_index = std::rand() % possible_candidates.size();
```

Comment:
For simulation results we require reproducability. We do not seed ``std::rand()``, ``std::rand`` can be effected by other libs  and the standard does not specify which algorithm is used [1]. 

For DAISI, ``daisi::global_random_engine`` is available as a random generator. This can be used like this with different distributions:
https://github.com/iml130/sola/blob/92109bd036e69f1e76a0f22f6b26e70f8e0810e7/daisi/src/natter-ns3/natter_manager.cpp#L207-L209

[1] https://stackoverflow.com/a/66903541